src/java.desktop/macosx/classes/sun/lwawt/LWWindowPeer.java

```diff
@@ -289,6 +289,10 @@ protected void setVisibleImpl(final boolean visible) {
         updateFocusableWindowState();
         super.setVisibleImpl(visible);
         // TODO: update graphicsConfig, see 4868278
+        if (visible) {
+            // Set correct background for a window before making it visible
+            platformWindow.setOpaque(!isTranslucent());
+        }
```

Comment:
This looks as if it will affect OpenGL as well as Metal.
Do you see the flckering with both pipelines ?
What testing have you done with both of them ?

Reply:
Yes, it affects both pipelines, but flickering was noticed only with Metal. I've tried IDEA on both pipelines and I've run desktop tests only with Metal. I'm going to run desktop ones with OGL too.

Reply:
OGL desktop tests are OK.

> This looks as if it will affect OpenGL as well as Metal.
> Do you see the flckering with both pipelines ?
> What testing have you done with both of them ?

Reply:
If the user will create a peer then change the background color, then show the window the flickering will occur. you need to override the LWWIndowPeer.setBackground and call the platformWindow.setBackground from there. The LWWIndowPeer.setBackground is also called during peer initialization so the code above will not be needed. 

But it will be good to root cause the problem if it is not related to the performance.